src/Core/Tool/Authorizer/PostsLockAuthorizer.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/**
+ * Ushahidi Post Authorizer
+ *
+ * @author     Ushahidi Team <team@ushahidi.com>
+ * @package    Ushahidi\Application
+ * @copyright  2014 Ushahidi
+ * @license    https://www.gnu.org/licenses/agpl-3.0.html GNU Affero General Public License Version 3 (AGPL3)
+ */
+
+namespace Ushahidi\Core\Tool\Authorizer;
+
+use Ushahidi\Core\Entity;
+use Ushahidi\Core\Entity\User;
+use Ushahidi\Core\Entity\Form;
+use Ushahidi\Core\Entity\FormRepository;
+use Ushahidi\Core\Entity\UserRepository;
+use Ushahidi\Core\Entity\PostRepository;
+use Ushahidi\Core\Tool\Authorizer;
+use Ushahidi\Core\Tool\Permissions\Acl;
+use Ushahidi\Core\Tool\Permissions\Permissionable;
+use Ushahidi\Core\Traits\AdminAccess;
+use Ushahidi\Core\Traits\OwnerAccess;
+use Ushahidi\Core\Traits\ParentAccess;
+use Ushahidi\Core\Traits\PrivAccess;
+use Ushahidi\Core\Traits\UserContext;
+use Ushahidi\Core\Traits\PrivateDeployment;
+use Ushahidi\Core\Tool\Permissions\AclTrait;
+use Ushahidi\Core\Entity\Permission;
+
+// The `PostAuthorizer` class is responsible for access checks on `Post` Entities
+class PostsLockAuthorizer implements Authorizer
+{
+	// The access checks are run under the context of a specific user
+	use UserContext;
+
+	// To check whether the user has admin access
+	use AdminAccess;
+
+	// To check whether user owns the webhook
+	use OwnerAccess;
+
+	// It uses `PrivAccess` to provide the `getAllowedPrivs` method.
+	use PrivAccess;
+
+	// It uses `PrivateDeployment` to check whether a deployment is private
+	use PrivateDeployment;
+
+    use AclTrait;
+
+
+    /* Authorizer */
+    public function isAllowed(Entity $entity, $privilege)
```

Comment:
This should really delegate to the post authorizer, and check the user has edit permissions for the post first.